src/bridge_iroad/src/iroad_closedloop.cpp

```diff
@@ -91,32 +102,49 @@ void Teleop_IRoad::udp_cb(const ros::TimerEvent& event) {
 
   Matrix<double,3,1> r_f;       // Front vector in body frame
   r_f << 0.0,1.0,0.0;
-  steer = rad2deg(acos(r_tb.dot(r_f)/d_t));
+  
+  /*steer = rad2deg(acos(r_tb.dot(r_f)/d_t)); //debug: steering setting is equal to angle between vehicle heading and goal direction
 
-  steer = min(steer, 30.0);
+  steer = min(steer, max_steer);
+  steer /= max_steer;
 
   if (r_tb(0,0) < 0) {
     steer = -steer;
   } else {
     // Carry On
   }
 
+  */
+
+  double alpha = acos(r_tb.dot(r_f)/d_t); //angle between vehicle heading and direction of goal (in radians)
+  if (r_tb(0,0) < 0){
+    alpha = -alpha; //accounting for the difference between left and right turns
+  }
+  //double delta; //angle between vehicle heading and desired vehicle heading at target (in radians), not yet implemented
+
+  double omega = ctrl_k2*alpha + ctrl_k1*sin(alpha)*cos(alpha); //feedback control law dictates this angular velocity value (in terms of the vehicle's heading) to target the goal location, without a specific final heading (with respect to the world frame)
+  //double omega = ctrl_k2*alpha + ctrl_k1*sin(alpha)*cos(alpha)*(alpha + ctrl_k3*delta)/alpha; //instantaneous angular velocity required by feedback control law, if goal position includes a target heading
+
+  steer = rad2deg(atan2(whlbase*omega,v_const)); //conversion of required angular velocity of vehicle to steering angle that will result in that angular velocity
+  steer /= max_steer; //rescale steer value to be a proportion of maximum extent of steering mechanism
```

Comment:
Split the calculation of the steering angle from the feedback control law into several steps and added some more comments.